src/voting-contracts/contracts/Strategies.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import "../node_modules/@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "./IStrategies.sol";
+
+contract Strategies is IStrategies {
+    IERC20 private contractInstance;
+    uint256 private minimumAmount;
+
+    error ERC20StrategyError(string reason);
+
+    constructor(address _contractAddress, uint256 _minimumAmount) {
+        contractInstance = IERC20(_contractAddress);
+        minimumAmount = _minimumAmount;
+    }
+
+    function getBalance(address _owner) private view returns (uint256) {
+        try contractInstance.balanceOf(_owner) returns (uint256 balance) {
+            return balance;
+        } catch {
+            revert ERC20StrategyError("Error fetching balance");
+        }
+    }
+
+    function gateStrategyMinFungibleBalance(address _owner)
+        public
+        view
+        returns (bool)
+    {
+        return (getBalance(_owner) >= minimumAmount);
+    }
+
+    function vanillaVotingStrategy(address _voter)
+        public
+        view
+        returns (uint256)
+    {
+        if (getBalance(_voter) > 0) {
+            return 1;
+        }
+        revert ERC20StrategyError("Balance must be greater than 0");
+    }
+
+    function fungibleBalanceVotingStrategy(address _voter)
+        public
+        view
+        returns (uint256)
+    {
+        uint256 _balance = getBalance(_voter);
+        if (_balance > 0) {
+            return _balance;
+        }
+        revert ERC20StrategyError("Balance must be greater than 0");
+    }
+
+    // NOTE: This implementation will be replaced by frontend depending on the strategies selected
+    function evaluateVotingPower(address _voter)
+        external
+        pure
+        returns (uint256)
+    {
+        return 1;
+    }
+
+    // NOTE: This implementation will be replaced by frontend depending on the strategies selectedx
+    function evaluateGatingStrategies(address _voter)
```

Comment:
`_voter` is not used. Inside strategies, `msg.sender` can be passed

Reply:
- @saad-s please let me know if we are using the `Strategies` contract in a proper way [here](https://github.com/saad-s/snapshot-y/blob/42179d2bac5c4b6a02eb6b3ce35b9c3bb8696e29/src/voting-contracts/contracts/Proposal.sol#L233). 
- If we don't pass the `_voter` the `Proposal` contract address will become `msg.sender`.
- In order to avoid passing this argument, we should use `tx.origin `. 
- The difference between` msg.sender` and `tx.origin` is explained[ here ](https://blockchain-academy.hs-mittweida.de/courses/solidity-coding-beginners-to-intermediate/lessons/solidity-5-calling-other-contracts-visibility-state-access/topic/tx-origin-and-msg-sender/)